### ISSUE TITLE
can't compile with cmake

### ISSUE THREAD

**Issue by themasch** [deoxxa/libmcnet#1]
Hi, 

i just tried to compile the develop-Branch with cmake . && make. 
This is the result: 

```
% LANG="C" make
[ 20%] Building C object CMakeFiles/mcnet.dir/src/metadata.c.o
[ 40%] Building C object CMakeFiles/mcnet.dir/src/parser.c.o
[ 60%] Building C object CMakeFiles/mcnet.dir/src/read.c.o
[ 80%] Building C object CMakeFiles/mcnet.dir/src/slot.c.o
Linking C static library libmcnet.a
[ 80%] Built target mcnet
[100%] Building C object CMakeFiles/example.dir/example.c.o
/home/masch/cpp/libmcnet/example.c: In function ‘on_packet’:
/home/masch/cpp/libmcnet/example.c:41:5: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 2 has type ‘int64_t’ [-Wformat]
Linking C executable example
CMakeFiles/example.dir/example.c.o: In function `on_packet':
example.c:(.text+0xa2): undefined reference to `mcnet_generator_size'
example.c:(.text+0x243): undefined reference to `mcnet_generator_size'
example.c:(.text+0x3cd): undefined reference to `mcnet_generator_size'
example.c:(.text+0x4a6): undefined reference to `mcnet_generator_size'
example.c:(.text+0x52e): undefined reference to `mcnet_generator_size'
CMakeFiles/example.dir/example.c.o:example.c:(.text+0x5f8): more undefined references to `mcnet_generator_size' follow
libmcnet.a(metadata.c.o): In function `mcnet_metadata_generator_write':
metadata.c:(.text+0x6f1): undefined reference to `mcnet_write_int16'
metadata.c:(.text+0x72a): undefined reference to `mcnet_write_int32'
metadata.c:(.text+0x763): undefined reference to `mcnet_write_float'
metadata.c:(.text+0x7a0): undefined reference to `mcnet_write_uint16'
metadata.c:(.text+0x81a): undefined reference to `mcnet_write_uint16'
metadata.c:(.text+0x841): undefined reference to `mcnet_write_uint16'
metadata.c:(.text+0x866): undefined reference to `mcnet_write_uint16'
metadata.c:(.text+0x89c): undefined reference to `mcnet_write_uint32'
metadata.c:(.text+0x8bd): undefined reference to `mcnet_write_uint32'
metadata.c:(.text+0x8de): undefined reference to `mcnet_write_uint32'
collect2: ld returned 1 exit status
make[2]: *** [example] Error 1
make[1]: *** [CMakeFiles/example.dir/all] Error 2
make: *** [all] Error 2
```

Looks like the lack of some -l or something like this?

**Comment by deoxxa** [comment]
I think I might have fixed this.

**Comment by deoxxa** [comment]
Yep this is fixed as of 5b7bd4a7b309c1f4e690c05b1118e724a8187279.
